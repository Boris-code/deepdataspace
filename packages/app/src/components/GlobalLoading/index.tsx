import { Spin } from 'antd';
import { ReactNode } from 'react';

interface IGlobalLoadingProps {
  children: ReactNode;
  active: boolean;
  tip?: string;
}

const GlobalLoading: React.FC<IGlobalLoadingProps> = ({ children, active }) => {
  return (
    <Spin
      size="large"
      spinning={active}
      delay={500}
      style={{
        zIndex: 99999,
        maxHeight: '100%',
        backgroundColor: '#E6F7FF',
        opacity: 0.2,
      }}
    >
      {children}
    </Spin>
  );
};

export default GlobalLoading;
