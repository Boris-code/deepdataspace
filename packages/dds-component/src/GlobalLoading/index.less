.dds-global-loading {
  z-index: 99999;
  max-height: '100%';
  background-color: '#E6F7FF';
  opacity: 0.2;
}
