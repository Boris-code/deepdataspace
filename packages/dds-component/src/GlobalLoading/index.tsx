import React from 'react';
import { Spin } from 'antd';
import './index.less';

interface IGlobalLoadingProps {
  children: React.ReactNode;
  active: boolean;
  tip?: string;
}

const GlobalLoading: React.FC<IGlobalLoadingProps> = ({ children, active }) => {
  return (
    <Spin
      size="large"
      spinning={active}
      delay={500}
      className="dds-global-loading"
    >
      {children}
    </Spin>
  );
};

export default GlobalLoading;
