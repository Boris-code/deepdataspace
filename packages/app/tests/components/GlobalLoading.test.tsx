import { render } from '@testing-library/react';
import GlobalLoading from '@/components/GlobalLoading';

test('render GlobalLoading by snapshot', () => {
  const { container } = render(<GlobalLoading active>test</GlobalLoading>);
  expect(container).toMatchSnapshot();
});
