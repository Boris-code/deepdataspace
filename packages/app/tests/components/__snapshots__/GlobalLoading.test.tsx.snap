// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render GlobalLoading by snapshot 1`] = `
<div>
  <div
    class="ant-spin-nested-loading"
  >
    <div
      class="ant-spin-container"
    >
      test
    </div>
  </div>
</div>
`;
